src/components/Button/TwoStateButton/TwoStateButton.tsx

```diff
@@ -0,0 +1,178 @@
+            },
+            { [styles.buttonPadding1]: size === ButtonSize.Large },
+            { [styles.buttonPadding2]: size === ButtonSize.Medium },
+            { [styles.buttonPadding3]: size === ButtonSize.Small },
+            { [styles.buttonStretch]: buttonWidth === ButtonWidth.fill },
+            { [styles.pillShape]: shape === ButtonShape.Pill },
+            { [styles.dropShadow]: dropShadow },
+            { [styles.left]: alignText === ButtonTextAlign.Left },
+            { [styles.right]: alignText === ButtonTextAlign.Right },
+            { [styles.disabled]: allowDisabledFocus || disabled },
+            { [styles.dark]: theme === ButtonTheme.dark },
+        ]);
+
+        const buttonTextClassNames: string = classNames([
+            { [styles.button3]: size === ButtonSize.Flex && largeScreenActive },
+            {
+                [styles.button2]:
+                    size === ButtonSize.Flex && mediumScreenActive,
+            },
+            { [styles.button2]: size === ButtonSize.Flex && smallScreenActive },
+            {
+                [styles.button1]:
+                    size === ButtonSize.Flex && xSmallScreenActive,
+            },
+            { [styles.button1]: size === ButtonSize.Large },
+            { [styles.button2]: size === ButtonSize.Medium },
+            { [styles.button3]: size === ButtonSize.Small },
+        ]);
+
+        const counterClassNames: string = classNames([
+            styles.counter,
+            buttonTextClassNames,
+        ]);
+
+        const getButtonIconSize = (): IconSize => {
+            let iconSize: IconSize;
+            if (size === ButtonSize.Flex && largeScreenActive) {
+                iconSize = IconSize.Small;
+            } else if (
+                size === ButtonSize.Flex &&
+                (mediumScreenActive || smallScreenActive)
+            ) {
+                iconSize = IconSize.Medium;
+            } else if (size === ButtonSize.Flex && xSmallScreenActive) {
+                iconSize = IconSize.Large;
+            } else if (size === ButtonSize.Large) {
+                iconSize = IconSize.Large;
+            } else if (size === ButtonSize.Medium) {
+                iconSize = IconSize.Medium;
+            } else if (size === ButtonSize.Small) {
+                iconSize = IconSize.Small;
+            }
+            return iconSize;
+        };
+
+        const getButtonIcon = (icon: IconName, color: string): JSX.Element => (
+            <Icon
+                className={styles.icon}
+                color={color}
+                path={icon}
+                size={getButtonIconSize()}
+            />
+        );
+
+        const getButtonText = (
+            buttonTextClassNames: string,
+            text: string
+        ): JSX.Element => (
+            <span className={buttonTextClassNames}>
+                {text ? text : 'Button'}
+            </span>
+        );
+
+        return (
+            <button
```

Comment:
Why arent we using BaseButton here ?

Reply:
BaseButton is already overloaded, better to keep these separate.